## *********************************************************
## 
## File autogenerated for the hre_random_node package 
## by the dynamic_reconfigure package.
## Please do not edit.
## 
## ********************************************************/

##**********************************************************
## Software License Agreement (BSD License)
##
##  Copyright (c) 2008, Willow Garage, Inc.
##  All rights reserved.
##
##  Redistribution and use in source and binary forms, with or without
##  modification, are permitted provided that the following conditions
##  are met:
##
##   * Redistributions of source code must retain the above copyright
##     notice, this list of conditions and the following disclaimer.
##   * Redistributions in binary form must reproduce the above
##     copyright notice, this list of conditions and the following
##     disclaimer in the documentation and/or other materials provided
##     with the distribution.
##   * Neither the name of the Willow Garage nor the names of its
##     contributors may be used to endorse or promote products derived
##     from this software without specific prior written permission.
##
##  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
##  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
##  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
##  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
##  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
##  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
##  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
##  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
##  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
##  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
##  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
##  POSSIBILITY OF SUCH DAMAGE.
##**********************************************************/

config_description = [{'srcline': 47, 'description': 'Desired operation mode', 'max': 1, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '../cfg/hre_random_alg_config.cfg', 'name': 'op_mode', 'edit_method': "{'enum_description': 'Possible operation modes.', 'enum': [{'srcline': 42, 'description': 'Execute random sequences', 'srcfile': '../cfg/hre_random_alg_config.cfg', 'cconsttype': 'const int', 'value': 0, 'ctype': 'int', 'type': 'int', 'name': 'Random'}, {'srcline': 43, 'description': 'Execute a given sequence', 'srcfile': '../cfg/hre_random_alg_config.cfg', 'cconsttype': 'const int', 'value': 1, 'ctype': 'int', 'type': 'int', 'name': 'Sequence'}]}", 'default': 0, 'level': 0, 'min': 0, 'type': 'int'}, {'srcline': 48, 'description': 'Sequence file to execute', 'max': '', 'cconsttype': 'const char * const', 'ctype': 'std::string', 'srcfile': '../cfg/hre_random_alg_config.cfg', 'name': 'seq_file', 'edit_method': '', 'default': '', 'level': 0, 'min': '', 'type': 'str'}, {'srcline': 49, 'description': 'Desired interval (in cicles) to execute the home sequence', 'max': 10, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '../cfg/hre_random_alg_config.cfg', 'name': 'home_interval', 'edit_method': '', 'default': 3, 'level': 0, 'min': 2, 'type': 'int'}, {'srcline': 50, 'description': 'Desired interval (in cicles) to speak', 'max': 10, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '../cfg/hre_random_alg_config.cfg', 'name': 'speech_interval', 'edit_method': '', 'default': 1, 'level': 0, 'min': 1, 'type': 'int'}, {'srcline': 51, 'description': 'Minimum interval (in cicles) between two identical motion sequences', 'max': 10, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '../cfg/hre_random_alg_config.cfg', 'name': 'repeat_interval', 'edit_method': '', 'default': 3, 'level': 0, 'min': 1, 'type': 'int'}]

min = {}
max = {}
defaults = {}
level = {}
type = {}
all_level = 0

for param in config_description:
    min[param['name']] = param['min']
    max[param['name']] = param['max']
    defaults[param['name']] = param['default']
    level[param['name']] = param['level']
    type[param['name']] = param['type']
    all_level = all_level | param['level']

HreRandom_Random = 0
HreRandom_Sequence = 1
